### PR TITLE
Cancel Payment command (again and again)

### PR DESCRIPTION
**Pull request by Gonals**
HOLD https://github.com/Expensify/Auth/pull/9797

### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/327308
$ https://github.com/Expensify/App/issues/34650
$ https://github.com/Expensify/App/issues/34676
PROPOSAL:

### Tests
**Pay elsewhere**
1. From an account (account A) create a new workspace and invite account B to it.
2. Log into account B and request money from the new workspace.
3. Log back into account A and Pay the money request. Select "Pay elsewhere" It should end up like this:
Screenshot 2023-12-13 at 5 22 55 PM

4. Confirm you now see the 3dot menu on the top right. Click on it:
Screenshot 2023-12-13 at 5 23 42 PM

5. Select the cancel payment option. A modal should appear. Confirm the cancellation:
Screenshot 2023-12-13 at 5 23 49 PM

7. The payment should now be cancelled and you should see a message indicating so. You should now be able to pay again (and you can repeat this process a bunch of times):
Screenshot 2023-12-13 at 5 25 16 PM

8. After paying the money request report, request money, again, from account B. The request should be created in a new report.

9. Cancel the payment in the old report. You should end up with two processing reports. Any new requests should go to the newer report.

10. Pay the newer report. The older one should remain "processing", and any new requests should go to it.

**Pay with expensify**
- Make sure your admin has a valid withdrawal bank account, and your user a valid deposit one
- Repeat the previous flow, but using "Pay with Expensify" instead of "Pay elsewhere"
- Confirm it basically behaves the same way

**Paid policies**
- Log into oldDot from account A and go to Settings > Workspaces. 
- Create a new `Collect` workspace and open it.
- Once it is open, follow [these steps](https://stackoverflowteams.com/c/expensify/questions/17518/17519#17519) to make it available in NewDot.
- Invite Account B to the workspace.
- Log into Account B in NewDot. You should have a room for the new `Collect` workspace. Request money there.
- Make sure your admin has a valid withdrawal bank account, and your user a valid deposit one
- Basically repeat the previous flow. Everything should behave more or less the same, with the difference that, after cancelling, the report should go back to an "Approved" state, instead of "Processing" (You should not be able to re-approve it and the "Approve" button shouldn't be visible).

- [x] Verify that no errors appear in the JS console

### Offline tests
Same as tests, but offline

### QA Steps
Same as tests

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

**Comment by Gonals** [comment]
@sobitneupane, this is on HOLD for the paid workspaces. You probably can't test that part anyways, so feel free to go ahead and start on this for free workspaces

**Comment by Gonals** [comment]
Removing HOLD for now, as you cn start testing with free plans @sobitneupane

**Comment by sobitneupane** [comment]
I could not pay the amount from account A.
Screenshot 2024-02-05 at 12 59 13

https://github.com/Expensify/App/assets/25876548/530e22b3-803c-4fd1-9f52-952476a36978

**Comment by Gonals** [comment]
> I could not pay the amount from account A. Screenshot 2024-02-05 at 12 59 13
> 
>  Untitled.mov 
> I managed to pay from the Workspace Report Page, but I couldn't cancel the payment with following console error.
> 
> Screenshot 2024-02-05 at 13 07 54
>  Screen.Recording.2024-02-05.at.13.04.56.mov

Ah, likely something changed after merging main. Taking a look

**Comment by Gonals** [comment]
@sobitneupane, I can't seem to reproduce the issue. I merged main again, can you give it another shot?

**Comment by sobitneupane** [comment]
Thanks for the update @Gonals. I cannot reproduce the issue any longer.

I was able to reproduce [this issue](https://github.com/Expensify/App/issues/34685).

https://github.com/Expensify/App/assets/25876548/38c500de-5151-4259-bacb-909f14376a54

I was also able to reproduce [this issue](https://github.com/Expensify/App/issues/34681) and [this issue](https://github.com/Expensify/App/issues/34676).

https://github.com/Expensify/App/assets/25876548/164d84bc-2ccd-466f-9546-786bc36a3549

**Comment by sobitneupane** [comment]
Is cancel payment only limited to workspace? I could not cancel payment for transaction between two users.
Screenshot 2024-02-05 at 18 43 50

**Comment by Gonals** [comment]
Ok! Issues fixed!
- Negative amount issue fixed
- We shouldn't even be displaying the "Marked as reimbursed" message, so removing it.
- Due to non-related changes, we now display the Report's title on unsettled reports, so we should no longer see the cancelled/owes change

**Comment by sobitneupane** [comment]
@Gonals I can still reproduce [this issue](https://github.com/Expensify/App/issues/34681). 
Screenshot 2024-02-08 at 17 29 10

It first shows following message in LHN.
Screenshot 2024-02-08 at 17 29 38

which changes to following message after refreshing.
Screenshot 2024-02-08 at 17 30 10

**Comment by sobitneupane** [comment]
I found another weird issue. I requested $100 from a workspace(with NPR as default currency) as a member. Then I change the default currency to USD. I paid the amount. Now the message should be either paid $100 or NPR (something) but it is paid NPR 100.0.

Steps to reproduce:
1. [Member]Go to an workspace(with default currency other than USD) as a member
2. [Member] Actions> Request money > change the currency to USD and add amount > Next > Request
3. [Admin] Change the default currency of the workspace to USD
4. [Admin] Pay the amount
5. Notice that the amount is in one currency with the symbol of another currency (for example:- NPR100 expected: $100 or NPR (something))

https://github.com/Expensify/App/assets/25876548/195c36f1-3de8-4b25-a3b2-9736a2d341a1

**Comment by Gonals** [comment]
> @Gonals I can still reproduce [this issue](https://github.com/Expensify/App/issues/34681). Screenshot 2024-02-08 at 17 29 10
> 
> It first shows following message in LHN. Screenshot 2024-02-08 at 17 29 38
> 
> which changes to following message after refreshing. Screenshot 2024-02-08 at 17 30 10

Taking another look! I wasn't able to see it happen, but it is likely due to me being in the `canUseReportFields` beta, which changes this behavior

**Comment by Gonals** [comment]
> I found another weird issue. I requested $100 from a workspace(with NPR as default currency) as a member. Then I change the default currency to USD. I paid the amount. Now the message should be either paid $100 or NPR (something) but it is paid NPR 100.0.
> 
> Steps to reproduce:
> 
> 1. [Member]Go to an workspace(with default currency other than USD) as a member
> 2. [Member] Actions> Request money > change the currency to USD and add amount > Next > Request
> 3. [Admin] Change the default currency of the workspace to USD
> 4. [Admin] Pay the amount
> 5. Notice that the amount is in one currency with the symbol of another currency (for example:- NPR100 expected: $100 or NPR (something))
> 
>  Screen.Recording.2024-02-08.at.17.53.37.mov

I'd say this is unrelated to this PR. This PR may have made this issue visible, as we were using the wrong function to display these messages before (I'm surprised no one fixed that earlier), but it is not really causing the issue

**Comment by Gonals** [comment]
> > @Gonals I can still reproduce [this issue](https://github.com/Expensify/App/issues/34681). Screenshot 2024-02-08 at 17 29 10
> > It first shows following message in LHN. Screenshot 2024-02-08 at 17 29 38
> > which changes to following message after refreshing. Screenshot 2024-02-08 at 17 30 10
> 
> Taking another look! I wasn't able to see it happen, but it is likely due to me being in the `canUseReportFields` beta, which changes this behavior

I think this is fixed now. I wasn't able to reproduce it consistently before, but I've tried a few rounds and it seems to work fine now.

**Comment by sobitneupane** [comment]
Thanks for the update @Gonals . I can no longer reproduce [this issue](https://github.com/Expensify/App/pull/34719#issuecomment-1933957332). I will perform more extensive tests and will update.

https://github.com/Expensify/App/assets/25876548/f5003701-eac3-4093-a703-d7a7780576ec

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>
<summary>Android: Native</summary>



https://github.com/Expensify/App/assets/25876548/e0b5a1ad-0e2e-4f43-8deb-7fa7ea019c96


</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/25876548/a6169d47-1d0f-4d51-9ee9-15a3f283c69c




</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/25876548/1387d45b-59fa-41a8-91bd-02e33f5a20d0


https://github.com/Expensify/App/assets/25876548/eb4a1f64-b801-4f13-907d-a13be07c86a9



</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/25876548/d578c459-3c99-4403-a0ee-b9e2cd8fcef6



</details>

**Review by sobitneupane — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

File: src/components/MoneyReportHeader.tsx
Comment (blimpich) [inline src/components/MoneyReportHeader.tsx:70]: I think this has happened, see https://github.com/Expensify/App/pull/34092. Is it possible to get rid of this?

**Review by blimpich — CHANGES_REQUESTED** [review]

**Review by blimpich — APPROVED** [review]

**Pull request merged**
